OracleGoldenGate/21c/bin/deployment-main.sh

```diff
@@ -125,34 +130,69 @@ function setup_deployment_directories() {
-                ;;
-
-            false)
-                ${RUN_COMMAND} || abort "${RUN_COMMAND} did not complete successfully!"
-                ;;
-            *)
-                abort "Unknown RUN_COMMAND_BACKGROUND state [${RUN_COMMAND_BACKGROUND}] for run_prestart_command: expected true/false"
-        esac
+##  r u n _ u s e r _ s c r i p t s
+##
+## Hook for launching custom scripts in the container before and after ogg start
+##     Default Values:
+##       - ${ABORT_ON_USER_SCRIPT_ERRORS} : true
+##       - ${SETUP_USER_SCRIPTS}          : "${OGG_HOME}/scripts/setup"
+##       - ${STARTUP_USER_SCRIPTS}        : "${OGG_HOME/scripts/startup}"
+##
+## Scripts are run lexicographically and recursively from the directories pointed to by:
+##      - ${SETUP_USER_SCRIPTS} are executed prior to any other steps in the boot sequence
+##      - ${STARTUP_USER_SCRIPTS} are executed after ogg/nginx startup
+##
+## When ${ABORT_ON_USER_SCRIPT_ERRORS} is true:
+##    Scripts must return a status of 0 on exit or the boot proccess will abort
+##
+## When ${ABORT_ON_USER_SCRIPT_ERRORS} is false:
+##    Scripts that return a non-zero status will issue a warning but will continue
+##    the boot process
+##
+function run_user_scripts {
+
+    local SCRIPTS_ROOT="${1}";
+    
+    [ -z "$SCRIPTS_ROOT" ] && {
+        printf "%s: No SCRIPTS_ROOT passed on, no scripts will be run\n" "${0}";
+        return 1;
+    } || {
+        if [ -d "$SCRIPTS_ROOT" ] && [ -n "$(ls -A "$SCRIPTS_ROOT")" ]; then
+            printf "Executing user defined scripts in: %s\n" "${SCRIPTS_ROOT}"
+        
+            for f in "${SCRIPTS_ROOT}"/*; do
+                [ -d "${f}" ] && {
+                    run_user_scripts "${f}"
+                } || {
+                    case "$f" in
+                        *.sh)
+                                printf "%s: running %s\n" "${0}" "${f}";
+                                . "$f" && state=$? || state=$?
+                                ;;
+                        *.sql)
+                                printf "%s: running %s\n" "${0}" "${f}";
+                                echo "exit" | LD_LIBRARY_PATH=$(dirname ${SQLPLUS}):${LD_LIBRARY_PATH} ${SQLPLUS} -s "/ as sysdba" @"$f" && state=$? || state=$?
+                                ;;
+                        *)      
+                                printf "%s: ignoring %s\n" "${0}" "${f}" && state=$? || state=$?
+                                ;;
+                    esac
+                    [ $state -ne 0 ] && {
+                        printf "#################\n"
+                        printf "## WARNING - %s: user script failed [exit code: %s]: %s\n" "${0}" "${state}" "${f}"
+                        printf "#################\n"
+                        [ "${ABORT_ON_USER_SCRIPT_ERRORS,,}" == "true" ] && {
+                            printf "Aborting startup! - Sleeping 10 seconds before exit\n"
+                            sleep 10;
```

Comment:
Sleep?  I no sleep!
- Gollum

What are you thinking here with the pause?

Reply:
In Docker, the logs for a container are available until the container is destroyed.

Regardless, if it's specific to your environment, it should be in a user script. That would follow the behavior in thee DBMS container.